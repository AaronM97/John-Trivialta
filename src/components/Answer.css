.white {
  background: #ffffff;
}
.green {
  background: #16db06;
}
.red {
  background: #ff2d2d;
}

.answer {
  cursor: pointer;
  margin: 22px;
}

.answer > p {
  display: inline;
  padding: 3px 15px;
}