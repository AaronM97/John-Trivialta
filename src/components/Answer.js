import React from 'react';
import './Answer.css';

export default class Answer extends React.Component {
    state = {
        background: this.props.background
    }

    // decodes answers
    decodeHtml(html) {
    const txt = document.createElement("textarea");
    txt.innerHTML = html;
    return txt.value;
    }

    // resets background of answers to white when you click next using the props from the parent state
    static getDerivedStateFromProps(nextProps, prevState) {
        if (nextProps.hasbeenclicked === false) {
            return {
                background: nextProps.background
            };
        }
        return null
    }

   checkAnswer (e) {
       if (!this.props.hasbeenclicked) {
           if (e.target.id === "true") {
               this.setState({background: "green"})
            }
            if (e.target.id === "false") {
                this.setState({background: "red"})
            }
        }
    }

    render() {
        return (
            <div className="answer">
            <p id={this.props.isCorrect} className={this.state.background} onClick={this.props.clicker} onClickCapture={this.checkAnswer.bind(this)}
            >{this.decodeHtml(this.props.text)}</p>
        </div>
    )
}
}