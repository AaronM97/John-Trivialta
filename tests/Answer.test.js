import React from 'react';
import ReactDOM from 'react-dom';
import Answer from '../src/components/Answer';
import Enzyme, { shallow, mount } from 'enzyme';

describe('renders Answer.js without crashing', () => {
    it('smoke test', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Answer />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('calls decodeHtml() on render', () => {
        // shallow render Answer and spy on decodeHtml()
        //   then forceUpdate so we can check that it's been called on (re)render
        const simFn = jest.fn();
        const wrapper = shallow(<Answer isCorrect={'true'} background={'white'} key={100} hasbeenclicked={simFn} text={'hello'} clicker={simFn} />);
        const spy = jest.spyOn(wrapper.instance(), 'decodeHtml');
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(1);
    })

    it('should call checkAnswer on click of p', () => {
        const simFn = jest.fn()
        const wrapper = shallow(<Answer isCorrect={'true'} background={'white'} key={100} hasbeenclicked={simFn} text={'hello'} clicker={simFn} />);

        const spy = jest.spyOn(wrapper.instance(), 'checkAnswer')
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.find('p').simulate("clickCapture");
        expect(spy).toHaveBeenCalledTimes(1);

    })

    // it('state turns green on click', () => {
    //     const simFn = jest.fn()
    //     const wrapper = mount(<Answer isCorrect={'true'} background={'white'} key={100} hasbeenclicked={false} text={'hello'} clicker={simFn} />);
    //     // const p = wrapper.find('p')
    //     // expect(p).toHaveLength(1);
    //     // p.simulate('clickCapture', {target: {id: "true"}});
    //     wrapper.setState({background: "green"}, () => {
    //         wrapper.forceUpdate()
    //         expect(wrapper.state('background')).toEqual('green')
    //     })
    // })
})
