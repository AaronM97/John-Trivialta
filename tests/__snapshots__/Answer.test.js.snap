// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Answer.js without crashing should render correctly 1`] = `
<div>
  <p
    onClickCapture={[Function]}
  >
    undefined
  </p>
</div>
`;
